packages/agents/src/tests-d/typed-use-agent.test-d.ts

```diff
@@ -0,0 +1,49 @@
+import type { Agent } from "../../dist";
+import { env } from "cloudflare:workers";
+import { useAgent } from "../react";
+
+declare class A extends Agent<typeof env, {}> {
+  prop: string;
+  f1: () => number;
+  f2: (a: string) => void;
+  f3: (a: number, b: string) => Promise<string>;
+  f4: (a?: string) => void;
+  f5: (a: string | undefined) => void;
+  f6: () => Promise<void>;
+}
+
+// @ts-expect-error state doesn't match type A state
+const a2 = useAgent<{ foo: 'bar' }, A>({
+  agent: 'test',
+});
+
+const a1 = useAgent<{}, A>({
```

Comment:
is there any way we can flip the order of this signature without a breaking change? so that it's either useAgent<State> or useAgent<MayAgent, State>

Reply:
Hmm, my first approach was without the `State` parameter, and I didn't revert those changes when I added it to the new overload function.

I'll revert those changes.

Reply:
I did it in fe46392d506410f12db4a8f769ff497851116749. Actually `State` could be infered from `Agent`, but it breaks `useAgent<State>` calls if I omit it or set a default type to it, so we'll need to explicitly provide both `Agent` and `State` types.